samples/ControlCatalog/Pages/SplitButtonPage.xaml

```diff
@@ -0,0 +1,40 @@
+<UserControl xmlns="https://github.com/avaloniaui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             x:Class="ControlCatalog.Pages.SplitButtonPage">
+
+  <UserControl.Resources>
+    <MenuFlyout x:Key="SharedMenuFlyout"
```

Comment:
I kept a shared MenuFlyout here because it illustrates a possible bug I need feedback on. When a MenuFlyout is shared among multiple SplitButtons, each of those SplitButtons will received the Opened event and update their pseudoclasses accordingly. This means all SplitButtons with a shared Flyout will have the backgrounds changed together. I'm not sure how to work-around this. Should it just be documented as a limitation?

Reply:
Flyout can be shared, but it will be visible at one place at once. Target property will have a reference to the control where it was shown. Same in UWP.
So, you can filter by Target property before updating internal control state.

Reply:
This also occurs in the UWP version, so its just a behavior of Flyouts. 
I haven't tried this and am just thinking aloud, but I wonder if a bool flag could be placed in the `Click` handler of the secondary button that says the flyout is opening on this button, and then checked in the FlyoutOpening event handler that would then apply the `:flyoutopen` pseudoclass if it's appropriate to do so

Reply:
Thanks, you both have some good ideas here and I'll try a few things out to see if it can be fixed.

Reply:
Should be resolved: I ended up using the Target property to filter the events -- thanks, I didn't think about that property. This is an elegant and self-contained solution that doesn't require new, shared variables to I went with it.